bigframes/core/compile/aggregate_compiler.py

```diff
@@ -202,44 +226,102 @@ def _(
 
 
 @compile_unary_agg.register
-def _(op: agg_ops.MaxOp, column: ibis_types.Column, window=None) -> ibis_types.Value:
+def _(
+    op: agg_ops.MaxOp,
+    column: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
+) -> ibis_types.Value:
     return _apply_window_if_present(column.max(), window)
 
 
 @compile_unary_agg.register
-def _(op: agg_ops.MinOp, column: ibis_types.Column, window=None) -> ibis_types.Value:
+def _(
+    op: agg_ops.MinOp,
+    column: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
+) -> ibis_types.Value:
     return _apply_window_if_present(column.min(), window)
 
 
 @compile_unary_agg.register
 @numeric_op
-def _(op: agg_ops.StdOp, x: ibis_types.Column, window=None) -> ibis_types.Value:
+def _(
+    op: agg_ops.StdOp,
+    x: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
+) -> ibis_types.Value:
     return _apply_window_if_present(cast(ibis_types.NumericColumn, x).std(), window)
 
 
 @compile_unary_agg.register
 @numeric_op
-def _(op: agg_ops.VarOp, x: ibis_types.Column, window=None) -> ibis_types.Value:
+def _(
+    op: agg_ops.VarOp,
+    x: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
+) -> ibis_types.Value:
     return _apply_window_if_present(cast(ibis_types.NumericColumn, x).var(), window)
 
 
 @compile_unary_agg.register
 @numeric_op
-def _(op: agg_ops.PopVarOp, x: ibis_types.Column, window=None) -> ibis_types.Value:
+def _(
+    op: agg_ops.PopVarOp,
+    x: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
+) -> ibis_types.Value:
     return _apply_window_if_present(
         cast(ibis_types.NumericColumn, x).var(how="pop"), window
     )
 
 
 @compile_unary_agg.register
 def _(
-    op: agg_ops.CountOp, column: ibis_types.Column, window=None
+    op: agg_ops.CountOp,
+    column: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
 ) -> ibis_types.IntegerValue:
     return _apply_window_if_present(column.count(), window)
 
 
 @compile_unary_agg.register
-def _(op: agg_ops.CutOp, x: ibis_types.Column, window=None):
+def _(
+    op: agg_ops.ArrayAggOp,
+    column: ibis_types.Column,
+    window=None,
+    order_by: typing.Sequence[ibis_types.Value] = [],
+) -> ibis_types.ArrayValue:
```

Comment:
Instead of registering as a UnaryAggOp, and having to modify every other operation, should ArrayAgg be its own aggregate expression class, maybe OrderedUnaryAggOp?

Reply:
To avoid modifying other operation, creating a new `compile_ordered_unary_agg` dispatcher and let `can_order_by` property of the `UnaryAggOp` decides the dispatcher.